test/stablecoin_converter.js

```diff
@@ -273,476 +285,535 @@ contract("StablecoinConverter", async (accounts) => {
+          solution.prices,
+          solution.tokenIdsForPrice,
+          { from: solver }
+        ),
         "Solutions are no longer accepted for this batch"
       )
     })
-    it("rejects solution submission after 4 minute deadline is over", async () => {
+    it("[Basic Trade] rejects solution submission after 4 minute deadline is over", async () => {
       const feeToken = await MockContract.new()
       const stablecoinConverter = await StablecoinConverter.new(2 ** 16 - 1, feeDenominator, feeToken.address)
-
       const erc20_2 = await MockContract.new()
+
       await feeToken.givenAnyReturnBool(true)
       await erc20_2.givenAnyReturnBool(true)
 
-      await stablecoinConverter.deposit(feeToken.address, basicTrade.deposits[0].amount, { from: basicTrade.deposits[0].user })
-      await stablecoinConverter.deposit(erc20_2.address, basicTrade.deposits[1].amount, { from: basicTrade.deposits[1].user })
-
       await stablecoinConverter.addToken(erc20_2.address)
-      const batchIndex = (await stablecoinConverter.getCurrentBatchId.call()).toNumber()
-
-      const orderId1 = await sendTxAndGetReturnValue(stablecoinConverter.placeOrder, basicTrade.orders[0].buyToken, basicTrade.orders[0].sellToken, batchIndex + 1, basicTrade.orders[0].buyAmount, basicTrade.orders[0].sellAmount, { from: basicTrade.orders[0].user })
-      const orderId2 = await sendTxAndGetReturnValue(stablecoinConverter.placeOrder, basicTrade.orders[1].buyToken, basicTrade.orders[1].sellToken, batchIndex + 1, basicTrade.orders[1].buyAmount, basicTrade.orders[1].sellAmount, { from: basicTrade.orders[1].user })
 
+      // Make deposits
+      for (const deposit of basicTradeCase.deposits) {
+        const tokenAddress = await stablecoinConverter.tokenIdToAddressMap.call(deposit.token)
+        await stablecoinConverter.deposit(tokenAddress, deposit.amount, { from: accounts[deposit.user] })
+      }
 
+      // Place orders
+      const batchIndex = (await stablecoinConverter.getCurrentBatchId.call()).toNumber()
+      const orderIds = []
+      for (const order of basicTradeCase.orders) {
+        orderIds.push(
+          await sendTxAndGetReturnValue(
+            stablecoinConverter.placeOrder,
+            order.buyToken,
+            order.sellToken,
+            batchIndex + 1,
+            order.buyAmount,
+            order.sellAmount,
+            { from: accounts[order.user] }
+          )
+        )
+      }
       const time_remaining = (await stablecoinConverter.getSecondsRemainingInBatch()).toNumber()
       await waitForNSeconds(time_remaining + 241)
 
-      const prices = basicTrade.solution.prices
-      const owner = basicTrade.solution.owners
-      const orderId = [orderId1, orderId2]
-      const volume = basicTrade.solution.volume
-      const tokenIdsForPrice = basicTrade.solution.tokenIdsForPrice
+      const solution = basicTradeCase.solutions[0]
 
       // Should be exactly one second past when solutions are being accepted.
```

Comment:
can we make sure that the auction index is really correct for this? Because an incorrect auction index would throw the same error message.
Could we do the 
```
const batchIndex = (await stablecoinConverter.getCurrentBatchId.call()).toNumber()
``` 
right before we submit?

Reply:
Sure... but the line you commented on does not seem to contain anything like this.